python/lsst/eotest/sensor/ptcTask.py

```diff
@@ -167,25 +168,54 @@ def _fit_curves(self, ptc_stats, sensor_id):
         output = EOTestResults(outfile, namps=len(ptc_stats))
         for amp in ptc_stats:
             mean, var = np.array(ptc_stats[amp][0]), np.array(ptc_stats[amp][1])
-            # Compute the median gain and remove outliers.
-            med_gain = np.median(mean/var)
-            frac_resids = np.abs((var - mean/med_gain)/var)
-            index = np.where(frac_resids < self.config.max_frac_offset)
-            # Least-squares fit to the brighter-fatter model.
-            p0 = 0, med_gain
+            index_old = []
+            index = list(np.where(mean < 4e4)[0])
+            count = 1
+            # Initial guess for BF coeff, gain, and square of the read noise
+            pars = 2.7e-6, 0.75, 25
             try:
-                results = scipy.optimize.leastsq(residuals, p0, full_output=1,
-                                                 args=(mean[index], var[index]))
-                pars, cov = results[:2]
+                while index != index_old and count < 10:
+                    results = scipy.optimize.leastsq(residuals, pars, full_output=1,
+                                                     args=(mean[index], var[index]))
+                    pars, cov = results[:2]
+                    sig_resids = residuals(pars, mean, var)
+                    index_old = deepcopy(index)
+                    index = list(np.where(np.abs(sig_resids) < sig_cut)[0])
```

Comment:
It works in python 2.7 but the line 
`while index != index_old and count < 10`
results in a warning if index is not identically the same as index_old:  "DeprecationWarning: elementwise == comparison failed; this will raise an error in the future."